### PR TITLE
ENH: enable weights in oneDAL Decision Forest [CPU]

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description
enable weights in oneDAL Decision Forest [CPU]

### CODE REVIEW

File: cpp/oneapi/dal/algo/decision_forest/detail/train_ops.hpp [hunk 1]
```diff
@@ -53,6 +53,10 @@ struct train_ops {
         if (input.get_data().get_row_count() != input.get_responses().get_row_count()) {
             throw invalid_argument(msg::input_data_rc_neq_input_responses_rc());
         }
+        if (input.get_weights().has_data() &&
+            input.get_data().get_row_count() != input.get_weights().get_row_count()) {
+            throw invalid_argument(msg::input_data_rc_neq_input_weights_rc());
```
Comment (makart19) [inline cpp/oneapi/dal/algo/decision_forest/detail/train_ops.hpp:58]: Please add corresponding badarg test.

File: cpp/oneapi/dal/algo/decision_forest/train_types.cpp [hunk 2]
```diff
@@ -46,8 +48,8 @@ using detail::v1::train_result_impl;
 namespace v1 {
 
 template <typename Task>
-train_input<Task>::train_input(const table& data, const table& responses)
-        : impl_(new train_input_impl<Task>(data, responses)) {}
+train_input<Task>::train_input(const table& data, const table& responses, const table& weights)
```
Comment (agorshk) [inline cpp/oneapi/dal/algo/decision_forest/train_types.cpp:51]: This is a change to public API and should be discussed before merging the PR.

**Review by makart19 — APPROVED** [review]

**Review by outoftardis — APPROVED** [review]
image

**Review by Vika-F — APPROVED** [review]
I am Ok with the changes.

**Pull request merged**